src/openms/source/TRANSFORMATIONS/FEATUREFINDER/MultiplexFiltering.cpp

```diff
@@ -452,13 +452,31 @@ namespace OpenMS
     return OpenMS::Math::pearsonCorrelationCoefficient(pattern1.begin(), pattern1.end(), pattern2.begin(), pattern2.end());
   }
 
+
   double MultiplexFiltering::getAveragineSimilarity(const vector<double>& pattern, double m) const
+
   {
     // construct averagine distribution
     IsotopeDistribution distribution;
     vector<double> averagine_pattern;
     distribution.setMaxIsotope(pattern.size());
-    distribution.estimateFromPeptideWeight(m);
+    if (averagine_type_ == "peptide")
+    {
+        distribution.estimateFromPeptideWeight(m);
+    }
+    else if (averagine_type_ == "RNA")
+    {
+        distribution.estimateFromRNAWeight(m);
+    }
+    else if (averagine_type_ == "DNA")
+    {
+        distribution.estimateFromDNAWeight(m);
+    }
+    else
+    {
+        distribution.estimateFromPeptideWeight(m); //TODO: Deal with this intelligently
```

Comment:
What do you have in mind? - Can you not simply throw a `InvalidParameter` exception?
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/include/OpenMS/CONCEPT/Exception.h#L348